src/protocol.ts

```diff
@@ -326,6 +326,26 @@ export interface QueryDisposeParams {
 export interface QueryDisposeResult {
 }
 
+// ------------------------------- < Query Rename Request > ------------------------------------
+
+export namespace QueryRenameRequest {
+	export const type = new RequestType<QueryRenameParams, QueryRenameResult, void, void>('query/rename');
+}
+
+/**
+ * Parameters to provide when renaming a query (during a save in Azure Data Studio)
```

Comment:
I still am not sure having the message tied to renaming is what we want to do (we can have that conversation in the other thread), but in general regardless we wouldn't need to say anything about saving in the comment here. Not all renames are saves.